webapp/src/EventListener/ParseFilesForPutListener.php

```diff
@@ -0,0 +1,42 @@
+<?php declare(strict_types=1);
+
+namespace App\EventListener;
+
+use Riverline\MultiPartParser\Converters\HttpFoundation;
+use Symfony\Component\EventDispatcher\Attribute\AsEventListener;
+use Symfony\Component\HttpFoundation\File\UploadedFile;
+use Symfony\Component\HttpKernel\Event\ControllerEvent;
+
+#[AsEventListener(priority: 1)]
+class ParseFilesForPutListener
+{
+    public function __invoke(ControllerEvent $event): void
+    {
+        // When we have a PUT request, the files on the request will not be filled, since PHP doesn't
+        // fill the $_FILES array. We can manually parse the request to fill it using a library.
+        // This code is based on https://github.com/symfony/symfony/issues/36409#issuecomment-612442260.
+        $request = $event->getRequest();
+        if ($request->isMethod('PUT')) {
+            $document = HttpFoundation::convert($request);
+            if (!$document->isMultiPart()) {
```

Comment:
what is happening here?

Reply:
On that line or the whole file? The whole file I tried to explain in the comment and that line... checks if this request was actually a multipart upload, which is how browsers / HTTP do file uploads.

Reply:
This line. When can this be a non-multipart request but end up here? And should this silently return in this case?

Reply:
If you do a JSON PUT for example? I will add a comment.

Reply:
Does this added comment make it more clear?